pilot/pkg/serviceregistry/kube/controller/network_test.go

```diff
@@ -77,9 +77,9 @@ func TestNetworkUpdateTriggers(t *testing.T) {
 		setGws(c.NetworkGateways())
 	})
 	expectGateways := func(t *testing.T, expectedGws int) {
+		// wait for a notification
+		assert.ChannelHasItem(t, notifyCh)
```

Comment:
Is notifyCh a must here? I do think we reply on `retry.UntilSuccessOrFail` to check the gw cb correctly executed.

Reply:
It depends on the number of retries for the tests. If we have the channel, almost no retries will be performed. An interesting observation is that running 1000 tests with the notifyCh channel takes ~15 seconds, whereas it takes ~90 seconds without the channel.

Reply:
That's weird, IMO the retry at most increase 10ms.

Both channel and retry need to wait for the `CB`.  If the cb will only be run once, the channel is the most efficient, if the cb should be run more times, then channel behaves same as the retry.

Reply:
I just checked with a new environment, which has around 6 retries without the channel, and it resulted in an increase of 60ms. This is consistent with your statement. I guess my Mac environment is causing the unstable number of retries.